coda/coda_mdstore/presentation.py

```diff
@@ -79,6 +81,33 @@ def getFileHandle(codaId, codaPath):
     )
 
 
+def file_chunk_generator(url):
+    """
+    Download a file and stream it
+    """
+    r = requests.get(url, stream=True)
+    if r.status_code != 200:
+        return
+    for chunk in r.iter_content(1024):
+        yield chunk
+
+
+def zip_file_streamer(urls, identifier):
+    """
+    Stream downloadable zipped file using zipstream
+    """
+    with zipstream.ZipFile(mode='w') as zip_obj:
+        for url in urls:
+            meta_id = identifier.split('/')[-1]
```

Comment:
Let's pull this out of the loop, we don't need to be repeating this processing for each url, since the identifier isn't changing.

Reply:
You could also pass the metaid part only to the `zip_file_streamer` function rather than the full identifier.

Reply:
Like @ldko was saying, since we now pass in `meta_id`, we can just get rid of the `identifier` arg.

Reply:
oh...yeah. Will get rid of it.